packages/loopring_v3.js/src/protocol_v3.ts

```diff
@@ -0,0 +1,232 @@
+import BN from "bn.js";
+const fs = require("fs");
+import Web3 from "web3";
+
+interface StakeAmounts {
+  exchangeStake: BN;
+  protocolFeeStake: BN;
+}
+
+/**
+ * Processes all data of the Loopring v3 protocol contract.
+ */
+export class ProtocolV3 {
+  private web3: Web3;
+
+  private loopringAbi: string;
+
+  private loopringAddress: string;
+  private loopring: any;
+  private owner: string;
+
+  private implementationManagerAddress: string;
+  private version: string;
+  private enabled: boolean;
+
+  private syncedToEthereumBlockIdx: number;
+
+  private stakeAmounts: { [key: number]: StakeAmounts } = {};
+
+  /**
+   * Initializes a Protocol
+   * @param   web3                           The web3 instance that will be used to get the necessary data from Ethereum
+   * @param   implementationManagerAddress   The address of the implementation manager
+   * @param   loopringAddress                The address of the protocol address
+   * @param   version                        The version of the protocol
+   */
+  public async initialize(
+    web3: Web3,
+    loopringAddress: string,
+    implementationManagerAddress: string,
+    version: string
+  ) {
+    this.web3 = web3;
+    this.loopringAddress = loopringAddress;
+    this.implementationManagerAddress = implementationManagerAddress;
+    this.version = version;
+    this.enabled = true;
+
+    this.syncedToEthereumBlockIdx = 0;
+
+    const ABIPath = "ABI/version30/";
+    this.loopringAbi = fs.readFileSync(ABIPath + "ILoopringV3.abi", "ascii");
+
+    this.loopring = new web3.eth.Contract(JSON.parse(this.loopringAbi));
+    this.loopring.options.address = this.loopringAddress.toLowerCase();
```

Comment:
seems this "options" is not used anywhere else.

Reply:
It's the address that is used for the web3 contract used to do the calls on the smart contract and get the events. It's used internally in the `web3.eth.Contract`.